### PR TITLE
Refactor error state checking

### PR DESCRIPTION
**Pull request by sawsa307**
Create function ValidateEndpoints and move endpoint validation logics into EndpointCalculator. Now it would correctly validate endpoints for L4 and L7 endpoint calculators. Validation for L4 endpoints is currently a no-op.

### CODE REVIEW

**Comment by sawsa307** [comment]
cc @mmamczur

File: pkg/neg/syncers/transaction.go [hunk 1]
```diff
@@ -259,16 +261,15 @@ func (s *transactionSyncer) syncInternalImpl() error {
 	}
 	endpointsData := negtypes.EndpointsDataFromEndpointSlices(endpointSlices)
 	targetMap, endpointPodMap, dupCount, err = s.endpointsCalculator.CalculateEndpoints(endpointsData, currentMap)
-	if valid, reason := s.isValidEPField(err); !valid {
```
Comment (mmamczur) [inline pkg/neg/syncers/transaction.go:262]: thanks for getting rid of this part as well, it looked super confusing with this isValidEPField potentially ignoring errors

**Comment by mmamczur** [comment]
LGTM, probably best choice that you move this validation to the calculator

File: pkg/neg/types/sync_results.go [hunk 2]
```diff
@@ -50,6 +50,17 @@ var (
 	ResultOtherError        = "OtherError"
 	ResultInProgress        = "InProgress"
 	ResultSuccess           = "Success"
+
+	ErrorStateReason = map[error]string{
```
Comment (swetharepakula) [inline pkg/neg/types/sync_results.go:54]: add a comment as to why we need this.

File: pkg/neg/types/sync_results.go [hunk 3]
```diff
@@ -50,6 +50,17 @@ var (
 	ResultOtherError        = "OtherError"
```
Comment (swetharepakula) [inline pkg/neg/types/sync_results.go:50]: all the results should be constants.

Comment (sawsa307) [reply pkg/neg/types/sync_results.go]: Updated. Thanks!

File: pkg/neg/types/sync_results.go (hunk 2 above)
Comment (sawsa307) [reply pkg/neg/types/sync_results.go]: Added. Thanks!

**Pull request merged**